docs/sources/lids/_DRAFT-Index-Gateway-Sharding.md

```diff
@@ -0,0 +1,99 @@
+
+This approach would solve the problem of horizontal scaling. However, it does not solve the problem of different tenant sizes. It also fails to ensure replication for availability when running a small number of instances, unless there is a fixed lower value for the RF. It also tends to over-shard data in large index gateway deployments.
+
+### Proposal 2: Fixed per-tenant replication factor
+
+Adding a random shard ID (e.g. `shard-0`, `shard-1`, ... `shard-n`) to the tenant ID allows to utilize a certain amount of `n` instances. The amount of shards can be implemented as a per-tenant override. However, this approach results in non-deterministic hash keys.
+
+### Proposal 3: Shard by index files
+
+In order to answer requests, the index gateway needs to download index files from the object storage, and since Loki builds a daily index file per tenant, these index files can be sharded evenly across all available index gateway instances. Each instance is then assigned a unique set of index files which it can answer metadata queries for.
+
+This means that the sharding key is the name of the file in object storage. While this name encodes both the tenant and the date, this is not strictly necessary. Such a sharding mechanism could shard any files from object storage across a set of instances of a ring.
+
+If the time range for the requested metadata is within a single day then a single index gateway instance can answer the metadata request.
+However, if a metadata request spans multiple days, also multiple index gateway instances are involved. There are two ways to solve this:
+
+#### A) Split and merge on client side
+
+The client resolves the necessary index files and their respective gateway instances. It splits the request into multiple sub-requests, executes them and merges them into a single result.
+
+**Pros:**
+* Only the minimum necessary amount of requests are performed.
+
+**Cons:**
+* The client requires information about how to split and merge requests.
+
+#### B) Split and merge on index gateway handler side
+
+The client can execute a request on any index gateway. This handler instance then identifies the index files that are involved, splits the query, and resolves the appropriate instances. Once it received the sub-queries it resembles the full response result and sends it back to the client.
+
+**Pros:**
+* Sharding is handled transparently to the client.
+* Clients can communicate with any instance of the index gateway ring.
+* Domain information about splitting and merging is kept within index gateway server implementation.
+
+Due to it's architectural advantages, option B is proposed.
+
+While this proposal suggests the biggest change to the index gateway's sharding mechanism in terms of effort and complexity, it also offers the biggest win in terms of scalability.
```

Comment:
I'm not sure this is that different. Either the querier|query-frontend act as a "client", scattering/gathering requests across idx-gw instances, or an individual idx-gw does. Theoretically it could reduce some gossip pressure if non idx-gw nodes don't need to be aware of idx-gw ring membership, but I'm not sure this is a current issue or if our memberlist library supports this easily.